server/bot_messages.go

```diff
@@ -93,3 +99,42 @@ func (p *Plugin) notifyUserTeamsPrimary(userID string) {
 		p.GetAPI().LogWarn("Failed to notify user is Teams primary", "user_id", userID, "error", err)
 	}
 }
+
+func (p *Plugin) SendWelcomeMessageWithNotificationAction(userID string) error {
+	if err := p.botSendDirectPost(
+		userID,
+		p.makeWelcomeMessageWithNotificationActionPost(),
+	); err != nil {
+		return errors.Wrapf(err, "failed to send welcome message to user %s", userID)
+	}
+
+	return nil
+}
+
+func (p *Plugin) makeWelcomeMessageWithNotificationActionPost() *model.Post {
+	msg := []string{
+		"**Welcome to the MS Teams integration!**",
+		"Enable notifications to get notified about any mentions you get in direct messages or group messages in MS Teams directly within Mattermost.",
+		fmt.Sprintf("![enable notifications picture](%s/static/enable_notifications.gif)", p.GetURL()),
+	}
+
+	return &model.Post{
+		Message: strings.Join(msg, "\n\n"),
+		Props: model.StringInterface{
+			"attachments": []*model.SlackAttachment{
+				{
+					Actions: []*model.PostAction{
+						{
+							Integration: &model.PostActionIntegration{
+								URL: fmt.Sprintf("%s/enable-notifications", p.GetURL()),
```

Comment:
Can we avoid the fully qualified URL (and the image proxy issues) by just using a relative URL instead? (Taking into account any subpath, of course.)

Reply:
I applied some changes to use the relative URL, seems to be working fine for both the gif and the button action. I was afraid I would get rejected because not starting with `http{s}://`.

off-topic, but I wish we had a "hooky" way to deal with this, so I could declare my 

```gp
post := &model.Post{
		Message: strings.Join(msg, "\n\n"),
		Props: model.StringInterface{
			"attachments": []*model.SlackAttachment{
				{
					Actions: []*model.PostAction{
						{
							ID: "enable_notifications", // <-- change here
							Integration: {Plugin: pluginID}, // <-- change here
							Name:  "Enable Notifications",
							Style: "primary",
							Type:  model.PostActionTypeButton,
						},
					},
				},
			},
		},
	}
}
```

and in my plugin:

```go
OnPostAction(id string, data model.PostActionRequest) {
	switch id {
	// ....
	}
}
```

So many ideas, so little time to do it all right now :D